tests/test_core/conftest.py

```diff
@@ -75,15 +16,66 @@ async def p9(request, tmpdir):
     await ctx.close()
 
 
+@pytest.fixture
+def nsname():
+    ns = uifname()
+    netns.create(ns)
+    with IPRoute(netns=ns) as ipr:
+        ipr.link('set', index=1, state='up')
+        ipr.poll(ipr.addr, 'dump', address='127.0.0.1', timeout=5)
+    yield ns
+    try:
+        netns.delete(ns)
+    except Exception:
+        pass
+
+
+@pytest.fixture
+def link(nsname):
+    ifname = uifname()
+    with IPRoute(netns=nsname) as ipr:
+        ipr.link('add', ifname=ifname, kind='dummy', state='up')
+        (link,) = ipr.poll(ipr.link, 'dump', ifname=ifname, timeout=5)
+        yield link
+        try:
+            ipr.link('del', index=link.get('index'))
+        except NetlinkError as e:
+            if e.code != errno.ENODEV:
+                raise
+
+
+@pytest.fixture
+def index(link):
```

Comment:
The name here is also too generic imo

Reply:
Would `test_if_index` work?

Reply:
i don't think it's necessary to put `test` in pytest fixture names (pytest itself does not do it). However the interface for which this fixture returns an ifindex is always a dummy, isn't it ? So maybe the fixtures could be named `dummy_ifindex`, `dummy_ifname`, `dummy_interface`... ?